data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/GrokMatchUtil.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.logstash.mapping;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GrokMatchUtil {
+    private static final String GROK_MATCH_PATTERN_REGEX = "(?<openingBrackets>%\\{)(?<grokSyntax>[^{}:]+)(?<separator1>:)?" +
+            "(?<grokSemantic>[^{}:]+)?(?<separator2>:)?(?<dataType>[a-zA-Z]+)?(?<closingBrackets>}(\\s+)?)";
+    private static final Pattern GROK_MATCH_PATTERN = Pattern.compile(GROK_MATCH_PATTERN_REGEX);
+
+    public static String convertGrokMatchPattern(final String matchPattern) {
+        Matcher grokMatchPatternMatcher = GROK_MATCH_PATTERN.matcher(matchPattern);
+        StringBuilder convertedGrokMatchPattern = new StringBuilder();
+        while(grokMatchPatternMatcher.find()) {
+            convertedGrokMatchPattern.append(grokMatchPatternMatcher.group("openingBrackets"));
```

Comment:
It would be better to define these (openiingBrackets, grokSyntax, etc) as constants.  That way there's only one place that they are defined.  That way if someone changes the regex pattern it doesn't break the code below.

Reply:
Thanks for the feedback Phill. I made the changes to the logic to remove regex groups as it is getting complex.